quiz/src/Reducer-context.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { createContext, useContext, useReducer } from "react";
+
+import { quizdata } from "./Data/getQuiz";
+import { Quizdata } from "./Data/quiz.types";
+
+type statustype = "starting" | "finished" | "Running";
+type UserState = {
+  _id: string;
+  name: string;
+  email: string;
+};
+
+type quizstate = {
+  user: UserState;
+  score: number;
+  status: statustype;
+  currentQsnNo: number;
+  currentquiz: string;
+  correct: number;
+  wrong: number;
+  data: Quizdata;
+  categorydata: string[];
+};
```

Comment:
You can move these types to a different file and keep all the types in a single file.